cli/main.rs

```diff
@@ -887,29 +887,70 @@ async fn run_with_watch(flags: Flags, script: String) -> Result<(), AnyError> {
     })
   };
 
+  struct ModuleExecutor {
+    worker: MainWorker,
+    pending_unload: bool,
+  }
+
+  impl ModuleExecutor {
+    pub fn new(worker: MainWorker) -> ModuleExecutor {
+      ModuleExecutor {
+        worker,
+        pending_unload: false,
+      }
+    }
+
+    pub async fn execute(
+      &mut self,
+      main_module: &ModuleSpecifier,
+    ) -> Result<(), AnyError> {
```

Comment:
This is a good solution, but please add some docstrings here - it's not immediately obvious why this is needed. I'd also suggest a change of name, maybe `FileWatcherModuleExecutor`?

Reply:
> FileWatcherModuleExecutor

As for the naming, wasn't the original intent but I'm thinking this can be a more generic thing.
Think it might be useful to hoist this in a follow up and use it everywhere where we do this lifecycle sequence which typically looks like:
- maybe_start_coverage_collection
- execute_module
- dispatch `load`
- run_event_loop
- dispatch 'unload'
- maybe_stop_coverage_collection

We repeat these steps in a few places that do the typical execution cycle.
But I'm leaving that for a (somewhat immediate) follow-up, first pass I'm just about fixing this bug.